xray/audit/python/python_test.go

```diff
@@ -58,3 +59,22 @@ func TestBuildPipenvDependencyList(t *testing.T) {
 	// Test sub child module
 	audit.GetAndAssertNode(t, childNode.Nodes, "ptyprocess:0.7.0")
 }
+
+func TestBuildPoetryDependencyList(t *testing.T) {
+	// Create and change directory to test workspace
+	_, cleanUp := audit.CreateTestWorkspace(t, "poetry-project")
+	defer cleanUp()
+	// Run getModulesDependencyTrees
+	rootNodes, err := BuildDependencyTree(pythonutils.Poetry)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.NotEmpty(t, rootNodes)
+
+	// Test child module
+	childNode := audit.GetAndAssertNode(t, rootNodes, "pytest:5.4.3")
+	// Test sub child module
+	transitiveChildNode := audit.GetAndAssertNode(t, childNode.Nodes, "packaging:21.3")
```

Comment:
```suggestion
if assert.NotNil(t, childNode) {
	transitiveChildNode := audit.GetAndAssertNode(t, childNode.Nodes, "packaging:21.3")
	......
```
so it won't panic if childnodes is nil